### ISSUE TITLE
After 20.3 upgrade, System Settings won't launch

### ISSUE THREAD

**Issue by spamegg1** [linuxmint/cinnamon#10579]
* Cinnamon version 5.2.7
 * Distribution - Mint 20.3 Una
 * Graphics hardware *and* driver used
➜ inxi -G
Graphics:  Device-1: Advanced Micro Devices [AMD/ATI] Picasso driver: amdgpu v: kernel
           Display: x11 server: X.Org 1.20.13 driver: amdgpu,ati unloaded: fbdev,modesetting,vesa resolution: 1920x1080~60Hz
           OpenGL: renderer: AMD Radeon Vega 11 Graphics (RAVEN DRM 3.35.0 5.4.0-92-generic LLVM 12.0.0) v: 4.6 Mesa 21.0.3
 * 64 bit
 * Attach ~/.xsession-errors, or /var/log/syslog
[xsession-errors.txt](https://github.com/linuxmint/cinnamon/files/7827871/xsession-errors.txt)
[syslog.txt](https://github.com/linuxmint/cinnamon/files/7827874/syslog.txt)

**Issue**
Just upgraded to Una 20.3 now. System Settings won't open (neither anything related to it: Applets, Desklets, Backgrounds, Themes, or anything under the "Preferences" section of the Menu.)

**Steps to reproduce**
Type `cinnamon-settings` in the Terminal, it says `No module named 'six'`. Same with `cinnamon-settings applets`, `cinnamon-settings themes` and so on.

**Expected behaviour**
System Settings should launch fine.

**Other information**
```
 ➜ pip3 install six
Defaulting to user installation because normal site-packages is not writeable
Requirement already satisfied: six in ./.local/lib/python3.8/site-packages (1.16.0)
 ➜ apt search python-six
i A python-six                                                   - Python 2 and 3 compatibility library (Python 2 interface)
v   python-six:i386                                              -
p   python-six-doc                                               - Python 2 and 3 compatibility library (documentation)
```

**Comment by leigh123linux** [comment]
Cinnamon block the use of pip for settings, try installing python3-six

**Comment by spamegg1** [comment]
```bash
 ➜ apt install python3-six
[sudo] password for bora:        
Reading package lists... Done
Building dependency tree       
Reading state information... Done
python3-six is already the newest version (1.14.0-2).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
```

**Comment by leigh123linux** [comment]
Can you post

```
ls -l $(which python3)
```

**Comment by spamegg1** [comment]
```bash
 ➜ ls -l $(which python3)
lrwxrwxrwx 1 root root 9 Dec 22  2020 /usr/bin/python3 -> python3.8
```

**Comment by leigh123linux** [comment]
Does import work?

```
python3
import six
```

**Comment by spamegg1** [comment]
It does:
```python3
 ➜ python3
Python 3.8.10 (default, Nov 26 2021, 20:14:08) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import six
>>> 
```

**Comment by spamegg1** [comment]
Some other info that might be relevant:
```bash
 ➜ apt policy python3-six
python3-six:
  Installed: 1.14.0-2
  Candidate: 1.14.0-2
  Version table:
 *** 1.14.0-2 500
        500 http://archive.ubuntu.com/ubuntu focal/main amd64 Packages
        500 http://archive.ubuntu.com/ubuntu focal/main i386 Packages
        100 /var/lib/dpkg/status

 ➜ apt version cinnamon
5.2.7+una
```

**Comment by spamegg1** [comment]
OK I solved the issue! `apt install --reinstall python3-six` did the trick! @leigh123linux Thanks for all your help!

**Comment by pavolgaj** [comment]
I had a similar problem but with module "requests". I used a solution from StackOverflow: https://stackoverflow.com/a/69646375. Everything works fine, now.

**Comment by spamegg1** [comment]
Yep someone else is having the same issue, but with the PIL module: https://forums.linuxmint.com/viewtopic.php?f=46&p=2119312

**Comment by starfoxx82** [comment]
Hi folks,

for me, the error will not gone. After i type `apt install --reinstall python3-six` i've got a new error. 

`
Traceback (most recent call last):
  File "/usr/share/cinnamon/cinnamon-settings/cinnamon-settings.py", line 743, in <module>
    window = MainWindow()
  File "/usr/share/cinnamon/cinnamon-settings/cinnamon-settings.py", line 316, in __init__
    for module in modules:
  File "/usr/share/cinnamon/cinnamon-settings/modules/cs_calendar.py", line 6, in <module>
    import pytz
ModuleNotFoundError: No module named 'pytz'
Error in sys.excepthook:
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/apport_python_hook.py", line 153, in apport_excepthook
    with os.fdopen(os.open(pr_filename,
FileNotFoundError: [Errno 2] Datei oder Verzeichnis nicht gefunden: '/var/crash/_usr_share_cinnamon_cinnamon-settings_cinnamon-settings.py.1000.crash'
`
Reinstall any other packages?

**Comment by spamegg1** [comment]
Here's some more: https://forums.linuxmint.com/viewtopic.php?f=47&t=364028
```bash
sudo apt-get reinstall python3-requests
sudo apt-get reinstall python3-six
sudo apt-get reinstall python3-tz
```

**Comment by rosyth** [comment]
It may be possible to make a quick 'hack' to fix this issue...  It worked for me with a pytz missing error.
In the file, "/usr/share/cinnamon/cinnamon-settings/cinnamon-settings.py", around line 7, I had the line, 
`util.strip_syspath_locals()`
Comment out this line.
I then have everything working again.  (I can't say without doing anything else, since I tried some of the various remedies suggested here and elsewhere before I tried this).
It's simple, I've not seen anything else going wrong so far. And it's easy to undo!!
